plugins/modules/ec2_key.py

```diff
@@ -319,16 +356,26 @@ def main():
     name = module.params['name']
     state = module.params.get('state')
     key_material = module.params.get('key_material')
-    force = module.params.get('force')
     key_type = module.params.get('key_type')
+    tags = module.params.get('tags')
+
+    result = {}
 
     if key_type:
         module.require_botocore_at_least('1.21.23', reason='to set the key_type for a keypair')
 
     if state == 'absent':
-        delete_key_pair(module, ec2_client, name)
+        result = delete_key_pair(module, ec2_client, name)
+
     elif state == 'present':
-        create_key_pair(module, ec2_client, name, key_material, force, key_type)
+        # check if key already exists
+        key = find_key_pair(ec2_client, name)
+        if key:
+            result = handle_existing_key_pair_update(module, ec2_client, name, key)
+        else:
+            result = create_new_key_pair(module, ec2_client, name, key_material, key_type, tags)
```

Comment:
I think the only use of module here is handling check_mode, might be better passing module.check_mode rather than module